client/me/purchases/cancel-purchase/domain-options.jsx

```diff
@@ -116,6 +121,62 @@ const CancelPurchaseDomainOptions = ( {
 		</div>
 	);
 
+	const IncludedDomainTransferPartialRefund = () => (
+		<p>
+			{ translate(
+				'You will receive a partial refund of %(refundAmount)s which is %(planCost)s for the plan ' +
+					'minus %(domainCost)s for the domain.',
+				{
+					args: {
+						domainCost: includedDomainTransfer.priceText,
+						planCost: planCostText,
+						refundAmount: purchase.refundText,
+					},
+				}
+			) }
+		</p>
+	);
+
+	const NonRefundableDomainTransferMessage = () => (
+		<div>
+			<p>
+				{ translate(
+					'This plan includes a domain transfer, %(domain)s, normally a %(domainCost)s purchase. ' +
+						'The domain transfer will not be removed along with the plan, to avoid any interruptions for your visitors.',
+					{
+						args: {
+							domain: includedDomainTransfer.meta,
+							domainCost: includedDomainTransfer.priceText,
+						},
+					}
+				) }
+			</p>
+			<IncludedDomainTransferPartialRefund />
+		</div>
+	);
+
+	const CancelableDomainTransferMessage = () => (
+		<div>
+			<p>
+				{ translate(
+					'It seems you have a pending domain transfer for the domain %(mappedDomain)s. ' +
+						'In order to receive a full refund, please cancel the domain transfer before requesting the cancellation of your plan.',
+					{
+						args: {
+							mappedDomain: includedDomainTransfer.meta,
+						},
+					}
+				) }
+			</p>
+			<IncludedDomainTransferPartialRefund />
+			<p>
+				{ translate(
+					'In some cases, a domain transfer has progressed too far to be canceled. Please contact support if you have questions about this'
```

Comment:
Minor but I think this sentence should end with a period.

```suggestion
					'In some cases, a domain transfer has progressed too far to be canceled. Please contact support if you have questions about this.'
```